R/helper-functions.R

```diff
@@ -368,14 +448,32 @@ createTreeViz <- function(clusters, counts) {
 #' @param treeViz TreeViz object
 #' @param top number of top genes to be calculated
 #' @return `TreeViz` Object with added top_variable_gene information in metadata
-#' 
+#'
 #' @import scran
 #' @export
 #'
-find_top_variable_genes <- function(treeviz, top=100) {
+find_top_variable_genes <- function(treeviz, top = 100) {
   dec.treeviz <- modelGeneVar(assays(treeviz)$counts)
   top_n <- getTopHVGs(dec.treeviz, n = top)
   metadata(treeviz)$top_variable <- top_n
   
   treeviz
 }
+
+#' Calculates `tsne` Dimensionality Reduction on `TreeViz` object. The result is added to reduced_dim slot in metadata
+#'
+#' @param treeViz TreeViz object
+#' @return `TreeViz` Object with added 'TSNE'tnse`information in reduced_dim slot of metadata
+#'
+#' @import scran
+#' @export
+#'
+calculate_tsne <- function(treeviz) {
```

Comment:
should we add dimensionality calculation for other methods ? UMAP ? PCA ?

Reply:
This is only when the user does not provide any dimensionality reductions, so I don't think TSNE is enough